src/core-sdk-tasks/ReplaceFilesWithSymbolicLinks.cs

```diff
@@ -51,52 +57,36 @@ public override bool Execute()
                 return false;
             }
 
+            if (!System.IO.Directory.Exists(LinkToFilesFrom))
+            {
+                Log.LogError($"'{LinkToFilesFrom}' does not exist.");
+                return false;
+            }
+
             // Find all non-empty, non-symbolic link files.
-            IEnumerable<FileInfo> fse = new FileSystemEnumerable<FileInfo>(
-                                            Directory,
-                                            (ref FileSystemEntry entry) => (FileInfo)entry.ToFileSystemInfo(),
-                                            new EnumerationOptions()
-                                            {
-                                                AttributesToSkip = FileAttributes.ReparsePoint,
-                                                RecurseSubdirectories = true
-                                            })
+            string[] files = new FileSystemEnumerable<string>(
+                                        Directory,
+                                        (ref FileSystemEntry entry) => entry.ToFullPath(),
+                                        new EnumerationOptions()
+                                        {
+                                            AttributesToSkip = FileAttributes.ReparsePoint,
+                                            RecurseSubdirectories = true
+                                        })
             {
                 ShouldIncludePredicate = (ref FileSystemEntry entry) => !entry.IsDirectory
                                                                         && entry.Length > 0
-            };
-
-            // Group them by file size.
-            IEnumerable<string?[]> filesGroupedBySize = fse.GroupBy(file => file.Length,
-                                                                    file => file.FullName,
-                                                                    (size, files) => files.ToArray());
+            }.ToArray();
 
-            // Replace files with same content with hard link.
-            foreach (var files in filesGroupedBySize)
+            foreach (var file in files)
             {
-                for (int i = 0; i < files.Length; i++)
+                string fileName = Path.GetFileName(file);
+
+                // Look for a file with the same name in LinkToFilesFrom
+                // and replace it with a symbolic link if it has the same content.
+                string targetFile = Path.Combine(LinkToFilesFrom, fileName);
```

Comment:
Is this supposed to work in recursive subdirectories?  The file search seems to be recursive, but this `targetFile` logic won't work correctly in that case.

Reply:
This matches the structure of the packs (files are layed out in nested directories), and the shared frameworks (files are directly in the shared framework directory).